### ISSUE TITLE
Library failed to initialize due to Aio issue

### ISSUE THREAD

**Issue by djj0809** [Azure/azure-kusto-python#312]
#### Code Sample, a copy-pastable example if possible

```python
# Your code here

from azure.kusto.data.exceptions import KustoServiceError

```
#### Problem description

This line of code starts to fail since 2.1.0 with following error. It is OK when I switched back to 2.0.0.

> Python 3.6.9 (default, Apr 18 2020, 01:56:04)
> [GCC 8.4.0] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> from azure.kusto.data.exceptions import KustoServiceError
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/usr/local/lib/python3.6/dist-packages/azure/kusto/data/__init__.py", line 5, in <module>
>     from .client import KustoClient, KustoConnectionStringBuilder, ClientRequestProperties
>   File "/usr/local/lib/python3.6/dist-packages/azure/kusto/data/client.py", line 22, in <module>
>     from .security import _AadHelper
>   File "/usr/local/lib/python3.6/dist-packages/azure/kusto/data/security.py", line 5, in <module>
>     from ._token_providers import *
>   File "/usr/local/lib/python3.6/dist-packages/azure/kusto/data/_token_providers.py", line 10, in <module>
>     from azure.identity.aio import ManagedIdentityCredential as AsyncManagedIdentityCredential
>   File "/usr/local/lib/python3.6/dist-packages/azure/identity/aio/__init__.py", line 7, in <module>
>     from ._credentials import (
>   File "/usr/local/lib/python3.6/dist-packages/azure/identity/aio/_credentials/__init__.py", line 7, in <module>
>     from .default import DefaultAzureCredential
>   File "/usr/local/lib/python3.6/dist-packages/azure/identity/aio/_credentials/default.py", line 14, in <module>
>     from .managed_identity import ManagedIdentityCredential
>   File "/usr/local/lib/python3.6/dist-packages/azure/identity/aio/_credentials/managed_identity.py", line 14, in <module>
>     from .._authn_client import AsyncAuthnClient
>   File "/usr/local/lib/python3.6/dist-packages/azure/identity/aio/_authn_client.py", line 20, in <module>
>     from azure.core.pipeline.transport import AioHttpTransport
> ImportError: cannot import name 'AioHttpTransport'
> 

#### Output of ``pip freeze`` 

<details>

[paste the output of ``pip freeze`` here below this line]

adal==1.2.1
APScheduler==3.6.0
asn1crypto==0.24.0
attrs==17.4.0
Automat==0.6.0
azure==4.0.0
azure-applicationinsights==0.1.0
azure-batch==4.1.3
azure-common==1.1.21
azure-core==1.12.0
azure-cosmosdb-nspkg==2.0.2
azure-cosmosdb-table==1.0.5
azure-datalake-store==0.0.45
azure-eventgrid==1.3.0
azure-graphrbac==0.40.0
azure-identity==1.5.0
azure-keyvault==1.1.0
azure-kusto-data==2.1.0
azure-loganalytics==0.1.0
azure-mgmt==4.0.0
azure-mgmt-advisor==1.0.1
azure-mgmt-applicationinsights==0.1.1
azure-mgmt-authorization==0.50.0
azure-mgmt-batch==5.0.1
azure-mgmt-batchai==2.0.0
azure-mgmt-billing==0.2.0
azure-mgmt-cdn==3.1.0
azure-mgmt-cognitiveservices==3.0.0
azure-mgmt-commerce==1.0.1
azure-mgmt-compute==4.6.2
azure-mgmt-consumption==2.0.0
azure-mgmt-containerinstance==1.5.0
azure-mgmt-containerregistry==2.8.0
azure-mgmt-containerservice==4.4.0
azure-mgmt-cosmosdb==0.4.1
azure-mgmt-datafactory==0.6.0
azure-mgmt-datalake-analytics==0.6.0
azure-mgmt-datalake-nspkg==3.0.1
azure-mgmt-datalake-store==0.5.0
azure-mgmt-datamigration==1.0.0
azure-mgmt-devspaces==0.1.0
azure-mgmt-devtestlabs==2.2.0
azure-mgmt-dns==2.1.0
azure-mgmt-eventgrid==1.0.0
azure-mgmt-eventhub==2.6.0
azure-mgmt-hanaonazure==0.1.1
azure-mgmt-iotcentral==0.1.0
azure-mgmt-iothub==0.5.0
azure-mgmt-iothubprovisioningservices==0.2.0
azure-mgmt-keyvault==1.1.0
azure-mgmt-loganalytics==0.2.0
azure-mgmt-logic==3.0.0
azure-mgmt-machinelearningcompute==0.4.1
azure-mgmt-managementgroups==0.1.0
azure-mgmt-managementpartner==0.1.1
azure-mgmt-maps==0.1.0
azure-mgmt-marketplaceordering==0.1.0
azure-mgmt-media==1.0.0
azure-mgmt-monitor==0.5.2
azure-mgmt-msi==0.2.0
azure-mgmt-network==2.7.0
azure-mgmt-notificationhubs==2.1.0
azure-mgmt-nspkg==3.0.2
azure-mgmt-policyinsights==0.1.0
azure-mgmt-powerbiembedded==2.0.0
azure-mgmt-rdbms==1.8.0
azure-mgmt-recoveryservices==0.3.0
azure-mgmt-recoveryservicesbackup==0.3.0
azure-mgmt-redis==5.0.0
azure-mgmt-relay==0.1.0
azure-mgmt-reservations==0.2.1
azure-mgmt-resource==2.2.0
azure-mgmt-scheduler==2.0.0
azure-mgmt-search==2.1.0
azure-mgmt-servicebus==0.5.3
azure-mgmt-servicefabric==0.2.0
azure-mgmt-signalr==0.1.1
azure-mgmt-sql==0.9.1
azure-mgmt-storage==2.0.0
azure-mgmt-subscription==0.2.0
azure-mgmt-trafficmanager==0.50.0
azure-mgmt-web==0.35.0
azure-nspkg==3.0.2
azure-servicebus==0.21.1
azure-servicefabric==6.3.0.0
azure-servicemanagement-legacy==0.20.6
azure-storage-blob==1.5.0
azure-storage-common==1.4.2
azure-storage-file==1.4.0
azure-storage-queue==1.4.0
blinker==1.4
certifi==2019.3.9
cffi==1.12.3
chardet==3.0.4
click==6.7
cloud-init==18.2
colorama==0.3.7
command-not-found==0.3
configobj==5.0.6
constantly==15.1.0
cryptography==2.1.4
decorator==4.4.2
distro-info==0.18
elasticsearch==7.0.2
elasticsearch-dsl==7.0.0
Flask==1.1.1
httplib2==0.9.2
hyperlink==17.3.1
idna==2.6
incremental==16.10.1
isodate==0.6.0
itsdangerous==1.1.0
Jinja2==2.10.1
jsonpatch==1.16
jsonpointer==1.10
jsonschema==2.6.0
keyring==10.6.0
keyrings.alt==3.0
language-selector==0.1
MarkupSafe==1.0
msal==1.9.0
msal-extensions==0.3.0
msrest==0.6.6
msrestazure==0.6.0
numpy==1.16.4
oauthlib==3.0.1
PAM==0.4.2
pandas==0.24.2
pipenv==2018.11.26
portalocker==1.7.1
py==1.9.0
pyasn1==0.4.2
pyasn1-modules==0.2.1
pycparser==2.19
pycrypto==2.6.1
pygobject==3.26.1
PyJWT==1.5.3
pyOpenSSL==17.5.0
pyserial==3.4
python-apt==1.6.0
python-dateutil==2.8.0
python-debian==0.1.32
pytz==2019.1
pyxdg==0.25
PyYAML==3.12
requests==2.18.4
requests-oauthlib==1.2.0
requests-unixsocket==0.1.5
retry==0.9.2
SecretStorage==2.3.1
service-identity==16.0.0
six==1.11.0
ssh-import-id==5.7
statsd==3.3.0
systemd-python==234
Twisted==17.9.0
tzlocal==1.5.1
ufw==0.35
unattended-upgrades==0.1
urllib3==1.22
virtualenv==16.4.3
virtualenv-clone==0.5.3
Werkzeug==0.15.6
zope.interface==4.3.2
</details>

**Comment by ChrisCoe** [comment]
I have a similar issue

**Comment by AsafMah** [comment]
Solved in #311
